.github/workflows/hermetic_library_generation.yaml

```diff
@@ -19,11 +19,25 @@ on:
 
 env:
   HEAD_REF: ${{ github.head_ref }}
+  REPO_FULL_NAME: ${{ github.event.pull_request.head.repo.full_name }}
 
 jobs:
+  # This step is meant to inline the the repo full name env var in the `jobs.if` entry
+  # of the next step "library_generation".
+  # Unfortunately, we can't directly reference `env` in jobs.if
+  # (see https://github.com/actions/runner/issues/2372)
+  # The chosen workaround is to declare an intermediate step that will output the env var
+  prepare-repo-full-name:
+    name: Prepare environment
+    runs-on: ubuntu-latest
+    outputs:
+      REPO_FULL_NAME: ${{ env.REPO_FULL_NAME }}
+    steps:
+      - run: exit 0 # no op - we just need to declare the env var as output
   library_generation:
-    # skip pull requests come from a forked repository
-    if: github.event.pull_request.head.repo.full_name == github.repository
+    # skip pull requests coming from a forked repository
+    if: needs.prepare-repo-full-name.outputs.REPO_FULL_NAME == github.repository
```

Comment:
What if we move this if condition to the `run` section? Did we consider the approach before @JoeWang1127 ?

Reply:
I modified it to just have an `if` inside `run`. I'll try a fork PR into this branch

Reply:
I tried raising [a PR from a fork](https://github.com/googleapis/java-bigtable/pull/2325)
![image](https://github.com/user-attachments/assets/b5cff9bf-af50-4e71-b9c9-23bef8399355)

I tried setting the missing token in my fork without luck.
However I could confirm that the generation was properly triggered in this PR because it doesn't come from a fork.
https://github.com/googleapis/java-bigtable/actions/runs/10562926017/job/29261969452?pr=2317